src/main/java/com/hierynomus/smbj/auth/NtlmAuthenticator.java

```diff
@@ -60,19 +61,24 @@ public NtlmAuthenticator create() {
         }
     }
 
-    public long authenticate(Connection connection, AuthenticationContext context) throws TransportException {
+    public Session authenticate(Connection connection, AuthenticationContext context) throws TransportException {
```

Comment:
I was trying not to rearchitect too much from your original design.  That said, I don't disagree with your comment.  But does that block the pull request, or can we deal with that later?